pkg/devicemapper/devmapper.go

```diff
@@ -358,12 +358,16 @@ func RemoveDeviceDeferred(name string) error {
 		return ErrTaskDeferredRemove
 	}
 
-	// cookie needed here, or else libdevmapper will disable udev dm rules
-	// and delete the symlink under /dev/mapper, even the device removal is deferred
-	// by the kernel, but no need to call UdevWait on the cookie after task.run,
-	// since deferred device removal is an asynchronous api call.
+	// set a task cookie and disable library fallback, or else libdevmapper will
+	// disable udev dm rules and delete the symlink under /dev/mapper by itself,
+	// no matter the removal is executed immediately or deferred by the kernel.
+	// Calling UdevWait on the cookie after task.run is not necessary, since deferred
```

Comment:
semaphore leak observed in my development environment, I think it is caused by not calling 'UdevWait'.So, plz DONT merge this PR for now!